File: backend/mallapi/src/main/java/com/nareun/service/TodoService.java
package com.nareun.service;

import com.nareun.dto.PageRequestDTO;
import com.nareun.dto.PageResponseDTO;
import com.nareun.dto.TodoDTO;

public interface TodoService {
    Long register(TodoDTO todoDTO);

    TodoDTO get(Long tno);

    void modify(TodoDTO todoDTO);

    void remove(Long tno);

    PageResponseDTO<TodoDTO> list(PageRequestDTO pageRequestDTO);
}

File: backend/mallapi-kotlin/src/main/kotlin/com/nareun/mallapikotlin/service/TodoService.kt
package com.nareun.mallapikotlin.service

import com.nareun.mallapikotlin.dto.PageRequestDTO
import com.nareun.mallapikotlin.dto.PageResponseDTO
import com.nareun.mallapikotlin.dto.TodoDTO

interface TodoService {
     fun register(todoDTO: TodoDTO) : Long?

     fun get(tno:Long) : TodoDTO

     fun modify(todoDTO: TodoDTO)

     fun remove(tno:Long)

     fun list(pageRequestDTO: PageRequestDTO): PageResponseDTO<TodoDTO>

}